lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
+  const fileRecords = await t.context.filePgModel.search(t.context.knex, {});
+  t.is(records.length, 1);
+  t.is(fileRecords.length, 0);
+
+  t.deepEqual(result, {
+    filesResult: {
+      total_dynamo_db_records: 0,
+      failed: 0,
+      skipped: 0,
+      migrated: 0,
+    },
+    granulesResult: {
+      total_dynamo_db_records: 1,
+      failed: 0,
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
+  t.teardown(async () => {
+    await t.context.granulePgModel.delete(t.context.knex, { cumulus_id: records[0].cumulus_id });
+    // shouldn't need to delete file...
+  });
+});
+
+test.serial('migrateGranuleAndFilesViaTransaction removes previously migrated files if a granule is re-migrated with an empty array of files', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  // migrate 1st time
+  await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+  });
+
+  // change api (dynamo) granule to have empty array
+  testGranule.files = [];
+
+  // migrate 2nd time
+  const result = await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+    migrateAndOverwrite: 'true',
+  });
+
+  const records = await t.context.granulePgModel.search(t.context.knex, {});
+  const fileRecords = await t.context.filePgModel.search(t.context.knex, {});
+  t.is(records.length, 1);
+  t.is(fileRecords.length, 0);
+
+  t.deepEqual(result, {
+    filesResult: {
+      total_dynamo_db_records: 0,
+      failed: 0,
+      skipped: 0,
+      migrated: 0,
+    },
+    granulesResult: {
+      total_dynamo_db_records: 1,
+      failed: 0,
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
+  t.teardown(async () => {
+    await t.context.granulePgModel.delete(t.context.knex, { cumulus_id: records[0].cumulus_id });
+    // shouldn't need to delete file...
+  });
+});
+
+test.serial('migrateGranuleAndFilesViaTransaction updates previously migrated files correctly if a granule is re-migrated with different files', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile1 = testGranule.files[0];
+  const fakeFile2 = fakeFileFactory({
+    bucket,
+    key: cryptoRandomString({ length: 10 }),
+    size: 1098034,
+    fileName: cryptoRandomString({ length: 20 }),
+    checksum: 'checkSum02',
+    checksumType: 'md5',
+    type: 'data',
+    source: 'source2',
+  });
+  testGranule.files.push(fakeFile2);
+
+  // migrate 1st time
+  const firstResult = await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+  });
+
+  t.deepEqual(firstResult, {
+    filesResult: {
+      total_dynamo_db_records: 2,
+      failed: 0,
+      skipped: 0,
+      migrated: 2,
+    },
+    granulesResult: {
+      total_dynamo_db_records: 1,
+      failed: 0,
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
```

Comment:
Can we check pg record after the first migration?

Reply:
added